src/pixelator/cli/collapse.py

```diff
@@ -236,38 +237,43 @@ def collapse(
                         )
                     )
 
+        total_input_reads = 0
         tmp_files = []
         for job in futures.as_completed(jobs):
             if job.exception() is not None:
                 raise job.exception()
             # the worker returns a path to a file (temp antibody edge list)
-            tmp_file = job.result()
+            tmp_file, input_reads_count = job.result()
             if tmp_file is not None:
                 tmp_files.append(tmp_file)
+            if input_reads_count is not None:
+                total_input_reads += input_reads_count
 
         # create the final edge list from all the temporary ones (antibodies)
         logger.debug("Creating edge list for sample %s", sample)
-        df = pd.concat(
-            (pd.read_feather(f, use_threads=True) for f in tmp_files), axis=0
+        df = pl.concat(
+            (pl.read_ipc(f, memory_map=False) for f in tmp_files), how="vertical"
         )
 
+        # Collect some stats for reporting
+        mean_collapsed_fragments = float(df["unique_fragment_count"].mean())
+        output_read_count = int(df["count"].sum())
+        molecule_count = int(df.shape[0])
```

Comment:
The collapsed_molecule_count_stats needs to remain there since it is dropped later, but I have moved the other two metrics closer to `CollapseStageReport` as suggested